docker/experimental/Terraform.md

```diff
@@ -0,0 +1,53 @@
+# Terraform
+
+This file contains instructions for common Terraform operations.
+
+## Install Terraform
+
+Make sure that a recent Terraform binary is installed (>= 1.3.8).
+If not, install Terraform from the [official source](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli).
+
+## First time initialization
+
+1. Run `gcloud auth application-default login` on your local workstation.
+2. Go to the directory of the desired Terraform setup, for example 
+   [`tpu-pytorch-releases`](./tpu-pytorch-releases).
+3. Run `terraform init`.
+
+## Enforce entire Terraform setup
```

Comment:
I'm still new to Terraform, so I want to understand better what a safe workflow looks like. We only want people to actually run `apply` after the changes are merged to `master`, right? Otherwise two people working on Terraform may overwrite each others' changes. Should the team even be running `terraform apply` themselves, or is that something we should build into our CI/CD pipeline?

Reply:
> Should the team even be running terraform apply themselves, or is that something we should build into our CI/CD pipeline?

Yeah we should, we could have a cloud build trigger that runs terraform apply.